cilium/tests/conftest.py

```diff
@@ -40,30 +42,45 @@ def setup_cilium():
 @pytest.fixture(scope='session')
 def dd_environment():
     kind_config = os.path.join(HERE, 'kind', 'kind-config.yaml')
+    use_openmetrics = False
+    if CILIUM_LEGACY == 'false':
+        use_openmetrics = True
     with kind_run(conditions=[setup_cilium], kind_config=kind_config) as kubeconfig:
         with ExitStack() as stack:
             ip_ports = [
                 stack.enter_context(port_forward(kubeconfig, 'cilium', port, 'deployment', 'cilium-operator'))
                 for port in PORTS
             ]
-        instances = {
-            'instances': [
-                {'agent_endpoint': 'http://{}:{}/metrics'.format(*ip_ports[0])},
-                {'operator_endpoint': 'http://{}:{}/metrics'.format(*ip_ports[1])},
-            ]
-        }
+
+            instances = {
+                'instances': [
+                    {
+                        'agent_endpoint': 'http://{}:{}/metrics'.format(*ip_ports[0]),
+                        'use_openmetrics': use_openmetrics,
+                    },
+                    {
+                        'operator_endpoint': 'http://{}:{}/metrics'.format(*ip_ports[1]),
+                        'use_openmetrics': use_openmetrics,
+                    },
+                ]
+            }
 
         yield instances
 
 
 @pytest.fixture(scope="session")
-def agent_instance():
-    return {'agent_endpoint': AGENT_URL, 'tags': ['pod_test']}
+def check():
+    return lambda instance: CiliumCheck('cilium', {}, [instance])
+
+
+@pytest.fixture(scope="session")
+def agent_instance_use_openmetrics():
+    return lambda legacy: {'agent_endpoint': AGENT_URL, 'tags': ['pod_test'], 'use_openmetrics': legacy}
```

Comment:
```suggestion
    return lambda use_openmetrics: {'agent_endpoint': AGENT_URL, 'tags': ['pod_test'], 'use_openmetrics': use_openmetrics}
```
nit since it should be set to false if it's legacy